core/src/components/badge/badge.ionic.scss

```diff
@@ -175,18 +182,33 @@
   height: globals.$ion-scale-400;
 }
 
+// Badge Sizes Not Empty With Vertical Prop
+// --------------------------------------------------
+
+:host(.badge-small[vertical]:not(:empty)),
+:host(.badge-medium[vertical]:not(:empty)),
+:host(.badge-large[vertical]:not(:empty)) {
+  @include globals.typography(globals.$ion-body-sm-medium);
+  min-width: globals.$ion-scale-400;
+  height: globals.$ion-scale-400;
+}
+
 // Badge Inside Tab Button
 // --------------------------------------------------
 
 :host([vertical].in-tab-button) {
   position: relative;
 }
 
-// Badge Icon when Inside Tab Button
-:host([vertical].in-tab-button) ::slotted(ion-icon) {
+// Icon Inside Badge Hint
+// --------------------------------------------------
+:host([vertical]) ::slotted(ion-icon) {
   @include globals.position(50%, null, null, 50%);
 
   position: absolute;
 
+  width: globals.$ion-scale-300;
+  height: globals.$ion-scale-300;
```

Comment:
This isn't necessary since `xxsmall` is the badge size to use to match the design. We already have the icon [set](https://github.com/ionic-team/ionic-framework/blob/ba3cf9c6d7f9b85923b02d47a4b4d399a800074f/core/src/components/badge/badge.ionic.scss#L63-L66) to use `scale-300`.

```suggestion
```

Reply:
Fixed!